validphys2/src/validphys/results.py

```diff
@@ -469,12 +492,55 @@ def results(dataset: (DataSetSpec), pdf: PDF, covariance_matrix, sqrt_covmat):
     The theory is specified as part of the dataset (a remnant of the old C++ layout)
     A group of datasets is also allowed.
     """
+    # TODO: is the message about the usage of the theory covariance matrix here true?
+    # probably not in most cases...
     return (
         DataResult(dataset, covariance_matrix, sqrt_covmat),
         ThPredictionsResult.from_convolution(pdf, dataset),
     )
 
 
+def results_with_theory_covmat(dataset, results, theory_covmat_dataset):
+    """Returns results with a modfy ``DataResult`` such that the covariance matrix includes
+    also the theory covmat.
+    This can be used to make use of results that consider scale variations without including
+    the theory covmat as part of the covariance matrix used by other validphys function.
+    Most notably, this can be used to compute the chi2 including theory errors while plotting
+    data theory covariance in which the experimental uncertainties are not stained by the thcovmat
+    """
+    # TODO: in principle this function could be removed, and `results` could automagically
```

Comment:
I guess you meant `automatically` but I like `automagically` a lot. It expresses very well how I fell when things work with `validphys` :)

Reply:
Nono, it is [100% intentional](https://www.oxfordlearnersdictionaries.com/definition/english/automagically)